### ISSUE TITLE
Not clear how to share attribute definitions between "attach" and "launch" configurations

### ISSUE THREAD

**Issue by gareth-rees** [microsoft/vscode#178138]
When writing an extension that needs custom launch configuration properties, the properties can be specified in package.json using the [debuggers contribution point](https://code.visualstudio.com/api/references/contribution-points#contributes.debuggers) and adding the schema of the custom properties under the `configurationAttributes` key.

When the extension needs to specify the same custom property for both the "launch" and "attach" configurations, there seems to be no way to share the property definition between these two cases, so that you have to copy and paste the definition into both configuration sections.

This was previously reported as issue #915, which was marked "closed as completed", but the [documentation for the debuggers contribution point](https://code.visualstudio.com/api/references/contribution-points#contributes.debuggers) does not explain how to do it. I suggest that the documentation be updated to explain how to share property definitions between "launch" and "attach" configurations, or if this is not possible, to add a note explaining that copy-pasting is necessary.

**Comment by roblourens** [comment]
I don't think that it was actually implemented, but I think we could support this.

**Comment by gareth-rees** [comment]
I have a feeling that it ought to be possible using [JSON schema references](https://json-schema.org/understanding-json-schema/structuring.html#ref). I couldn't get it to work in the context of the `configurationAttributes` section of package.json, but maybe I put the `$defs` and `$ref` in the wrong places. If it is possible, I think an example in the documentation would be really helpful.
